src/decorators/log.decorator.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback4-example-log-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Reflector, Constructor} from '@loopback/context';
+import {LogBindings, LogLevel} from '../keys';
+import {LogMetadata} from '../types';
+
+/**
+ * Mark a controller method as requiring logging (input, output & timing)
+ * if Application LogLevel is set at or greater than Application LogLevel.
+ * LogLevel.DEBUG < LogLevel.INFO < LogLevel.WARN < LogLevel.ERROR < LogLevel.OFF
+ *
+ * @param level The Log Level at or above it should log
+ */
+export function log(level?: number) {
+  return function(
+    target: Object,
+    methodName: string,
+    descriptor: PropertyDescriptor,
+  ) {
+    return {
+      value: function() {
+        // If a default log level isn't specified, we default to warn
+        if (!level) level = LogLevel.WARN;
+
+        // Take arguments array and change to array
+        const args = Array.prototype.slice.call(arguments);
```

Comment:
AFAICT, `LogMetadata.args` is not used anywhere - can we remove it?

I have mixed feelings about saving `arguments` into metadata. In order to obtain metadata, we already need to know at least "target" and "methodName", so what's the point of storing this information in the metadata again?